crates/codegen/parser/runtime/src/support/scanner_macros.rs

```diff
@@ -108,24 +108,6 @@ macro_rules! scan_optional {
     }};
 }
 
-#[allow(unused_macros)]
-macro_rules! scan_difference {
-    ($stream:ident, $minuend:expr, $subtrahend:expr) => {{
-        let start = $stream.position();
-        ($minuend)
-            && ({
-                let end = $stream.position();
-                $stream.set_position(start);
-                if ($subtrahend) && (end == $stream.position()) {
-                    false
-                } else {
-                    $stream.set_position(end);
-                    true
-                }
-            })
-    }};
-}
-
 #[allow(unused_macros)]
 macro_rules! scan_not_followed_by {
     ($stream:ident, $scanner:expr, $not_followed_by:expr) => {
```

Comment:
nit: I like how `delimited by`  tokens improve legibility of the DSL, maybe changing it to
```
($stream:ident, $scanner:expr, not followed by: $not_followed_by:expr)
```

would improve legibility of `scan_not_followed_by` macro calls here?